gateway/handlers/forwarding_proxy.go

```diff
@@ -131,7 +141,16 @@ func getServiceName(urlValue string) string {
 	var serviceName string
 	forward := "/function/"
 	if strings.HasPrefix(urlValue, forward) {
-		serviceName = urlValue[len(forward):]
+		// With a path like `/function/xyz/rest/of/path?q=a`, the service
+		// name we wish to locate is just the `xyz` portion.  With a postive
+		// match on the regex below, it will return a three-element slice.
+		// The item at index `0` is the same as `urlValue`, at `1`
+		// will be the service name we need, and at `2` the rest of the path.
+		matcher := functionMatcher.Copy()
+		matches := matcher.FindStringSubmatch(urlValue)
+		if 3 == len(matches) {
```

Comment:
As in Alex's comment bellow, let's not use a number for this check, but edit for better readability. 
One may be confused why there's a check for length of 3 to take the 2nd element and may edit and break it in future.